### DISCUSSION TITLE
Is there any synchronous (non-async) http mock for ureq testing?

### DISCUSSION THREAD

**Discussion by asv7c2** [algesten/ureq#873] [Q&A] [answered]
Or maybe i can somehow can mock ureq responses?

**Comment by algesten** [comment, marked as answer]
For ureq 2.x, it's a bit limited, but you got this constructor: https://docs.rs/ureq/2.10.1/ureq/struct.Response.html#method.new

For ureq 3.x the API is the [`http crate`](https://docs.rs/http/1.1.0/http/), and the body can be created any way you want: https://docs.rs/ureq/3.0.0-rc2/ureq/struct.Body.html#method.builder
